zyte_common_items/_addon.py

```diff
@@ -0,0 +1,34 @@
+from itemadapter import ItemAdapter
+from scrapy.settings import BaseSettings
+from scrapy.utils.misc import load_object
+
+from . import ZyteItemAdapter, ZyteItemKeepEmptyAdapter
+from .log_formatters import ZyteLogFormatter
+
+
+def _setdefault(settings, setting, cls, pos):
+    setting_value = settings[setting]
+    if not setting_value:
+        settings[setting] = {cls: pos}
+        return
+    if cls in setting_value:
+        return
+    for cls_or_path in setting_value:
+        if isinstance(cls_or_path, str):
+            _cls = load_object(cls_or_path)
+            if _cls == cls:
+                return
+    settings[setting][cls] = pos
+
+
+class Addon:
+    def update_settings(self, settings: BaseSettings) -> None:
+        if not any(
+            issubclass(cls, (ZyteItemAdapter, ZyteItemKeepEmptyAdapter))
+            for cls in ItemAdapter.ADAPTER_CLASSES
+        ):
+            ItemAdapter.ADAPTER_CLASSES = (ZyteItemAdapter,) + tuple(
+                ItemAdapter.ADAPTER_CLASSES
+            )
```

Comment:
I think we might need to convert it to deque, because itemadapter docs [suggest](https://github.com/scrapy/itemadapter?tab=readme-ov-file#registering-an-adapter) using .appendleft: any code which tries to register an adapter class after the addon is executed would break, if it follows itemadapter docs.

That's too bad we're making the change globally - e.g. it's not possible to run 2 spiders in the same proces, with different adapter classes enabled. But that's definitely out of scope here :)